dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java

```diff
@@ -53,33 +54,47 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
             return;
         }
 
-        request.setRequestUri(uri);
-        PageMode mode = PageMode.getWithNavigateMode(request);
-
-        try {
-            VelocityModeHandler.modeHandler(mode, request, response).serve();
-        } catch (ResourceNotFoundException rnfe) {
-            Logger.error(this, "ResourceNotFoundException" + rnfe.toString(), rnfe);
-            response.sendError(HttpServletResponse.SC_NOT_FOUND);
-        } catch (ParseErrorException pee) {
-            Logger.error(this, "Template Parse Exception : " + pee.toString(), pee);
-            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Template Parse Exception");
-        } catch (MethodInvocationException mie) {
-            Logger.error(this, "MethodInvocationException" + mie.toString(), mie);
-            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "MethodInvocationException Error on template");
-        } catch (Exception e) {
-            Logger.error(this, "Exception" + e.toString(), e);
-            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Exception Error on template");
+        final boolean comeFromAdmin = req.getAttribute(WebKeys.COME_FROM_ADMIN) != null ?
+                (boolean) req.getAttribute(WebKeys.COME_FROM_ADMIN) : false;
+
+        response.setHeader("Cache-Control", "no-cache, no-store, must-revalidate");
```

Comment:
Do we need to always disable the browser cache?, maybe we should do it only when editing the page and if it is a regular browsing allow the front end to interact with the browser normally.